pandas/core/ops.py

```diff
@@ -1078,15 +1132,6 @@ def wrapper(left, right):
                                     index=left.index, name=res_name,
                                     dtype=result.dtype)
 
-        elif is_categorical_dtype(left):
-            raise TypeError("{typ} cannot perform the operation "
-                            "{op}".format(typ=type(left).__name__, op=str_rep))
-
-        elif (is_extension_array_dtype(left) or
-              (is_extension_array_dtype(right) and
-               not is_categorical_dtype(right))):
```

Comment:
Is it obvious why the `is extension_array_dtype(right) and not is_categorical_dtype(right)` is dropping the second part of that check?

Reply:
i changed to catch up higher (where we already dispatch on categorical types)

Reply:
@jreback If you're making the change in the wrapper in `_arith_method_SERIES()` to change when you test for categorical and EA types, then you may need similar changes in the wrapper in `_comp_method_SERIES()`.

Reply:
@jreback see comment above - the dispatching to `dispatch_to_extension_op` is not consistent between `_arith_method_SERIES()` and `_comp_method_SERIES()`.